File: Python/Car.py
from Vehicle import Vehicle


class Car(Vehicle):
    def __init__(self,vehicle_id:str, model:str, fuel_level :float,fuel_efficiency :float):
        super().__init__(vehicle_id, model, fuel_level)
        self.fuel_efficiency = fuel_efficiency

    def calculate_range(self):
        range = self.fuel_level * self.fuel_efficiency
        print(f"Car {self.model} can travel {range} km with current fuel level.")
        return range
    

        
    

File: php/Car.php

<?php
include "Vehicle.php";
class Car extends Vehicle{
    private float $fuelEfficiency;

    public function __construct(string $vehicleId, string $model,$fuelLevel,$fuelEfficiency){
        parent::__construct($vehicleId, $model,$fuelLevel);
        $this->fuelEfficiency = $fuelEfficiency;
    }
public function getEfuelEfficiency() :float {
    return $this->fuelEfficiency;
    }
public function setEfuelEfficiency($fuelEfficiency) : void {
        $this->fuelEfficiency = $fuelEfficiency;
    }

public function calculateRange(): string {
        return $this->fuelEfficiency * $this-> fuelLevel;      
}
}
























?>